lib/pushmi_pullyu/cli.rb

```diff
@@ -160,15 +130,54 @@ def run_tick_loop
     end
   end
 
+  def running?
+    @running
+  end
+
+  def setup_log
+    if config.daemonize
+      PushmiPullyu::Logging.initialize_logger(config.logfile)
+    else
+      logger.formatter = PushmiPullyu::Logging::SimpleFormatter.new
+    end
+    logger.level = ::Logger::DEBUG if config.debug
+  end
+
+  def setup_signal_traps
+    Signal.trap('INT') { Thread.new { shutdown } }
+    Signal.trap('TERM') { Thread.new { shutdown } }
+    Signal.trap('HUP') { Thread.new { PushmiPullyu::Logging.reopen } }
```

Comment:
Comments got squashed after merge: https://github.com/ualbertalib/pushmi_pullyu/pull/44#discussion_r114667160

But agree, we will do this smarter